src/cdk-experimental/column-resize/overlay-handle.ts

```diff
@@ -0,0 +1,195 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+import {AfterViewInit, ElementRef, OnDestroy, NgZone} from '@angular/core';
+import {coerceCssPixelValue} from '@angular/cdk/coercion';
+import {Directionality} from '@angular/cdk/bidi';
+import {ESCAPE} from '@angular/cdk/keycodes';
+import {CdkColumnDef} from '@angular/cdk/table';
+import {fromEvent, ReplaySubject} from 'rxjs';
+import {
+  distinctUntilChanged,
+  filter,
+  map,
+  mapTo,
+  pairwise,
+  startWith,
+  takeUntil,
+} from 'rxjs/operators';
+
+import {closest} from '../popover-edit/polyfill';
+
+import {HEADER_CELL_SELECTOR} from './constants';
+import {ColumnResizeNotifierSource} from './column-resize-notifier';
+import {HeaderRowEventDispatcher} from './event-dispatcher';
+import {ResizeRef} from './resize-ref';
+
+export abstract class ResizeOverlayHandle implements AfterViewInit, OnDestroy {
+  protected readonly destroyed = new ReplaySubject<void>();
+
+  protected abstract readonly columnDef: CdkColumnDef;
+  protected abstract readonly document: Document;
+  protected abstract readonly directionality: Directionality;
+  protected abstract readonly elementRef: ElementRef;
+  protected abstract readonly eventDispatcher: HeaderRowEventDispatcher;
+  protected abstract readonly ngZone: NgZone;
+  protected abstract readonly resizeNotifier: ColumnResizeNotifierSource;
+  protected abstract readonly resizeRef: ResizeRef;
+
+  ngAfterViewInit() {
+    this._listenForMouseEvents();
+  }
+
+  ngOnDestroy() {
+    this.destroyed.next();
+    this.destroyed.complete();
+  }
+
+  private _listenForMouseEvents() {
```

Comment:
This function is pretty large- could parts of it be broken out into other methods?